src/content/docs/en/guides/migrate-to-astro/from-nuxtjs.mdx

```diff
@@ -2,68 +2,671 @@
+---
+<a href={to}><slot /></a>
+```
+
+### Nuxt Imports to Astro
+
+If necessary, update any [file imports](/en/guides/imports/) to reference relative file paths exactly. This can be done using [import aliases](/en/guides/typescript/#import-aliases), or by writing out a relative path in full. 
+
+Note that `.astro` and several other file types must be imported with their full file extension.
+
+```astro title="src/pages/authors/Fred.astro" ".astro"
+---
+import Card from `../../components/Card.astro`;
+---
+<Card />
+```
+
+### Nuxt Dynamic Page Generation to Astro
+
+In Nuxt, to generate a dynamic page you either must:
+
+- Use SSR
+- [Use the `generate` function in `nuxt.config.js` to define all possible static routes](https://nuxtjs.org/docs/configuration-glossary/configuration-generate/)
+
+In Astro, you must export a `getStaticPaths` function to tell the framework which routes it is expected to generate for a dynamic page.
 
-- Add your own!
+Assuming you have a Nuxt page called `/pages/pokemon/_name.vue` that you want to generate for each Pokémon, you would have the following function in your `nuxt.config.js`:
+
+```javascript title="nuxt.config.js"
+{
+	// ...
+    generate: {
+        async routes() {
+          // Axios is required here unless you're using Node 18
+          const res = await axios.get("https://pokeapi.co/api/v2/pokemon?limit=151")
+          const pokemons = res.data.results;
+
+          return pokemons.map(pokemon => {
+            return '/pokemon/' + pokemon.name
+          })
+        }
+      }
+}
+```
+
+This is replaced in Astro with a function that's located within the code block of your page itself:
+
+```astro title="src/pages/pokemon/[name].astro"
+---
+export const getStaticPaths = async () => {
+  const res = await fetch("https://pokeapi.co/api/v2/pokemon?limit=151")
+  const resJson = await res.json();
+  const pokemons = resJson.results;
+  return pokemons.map(({ name }) => ({
+      params: { name },
+    }))
+}
+
+// ...
+---
+
+<!-- Your template here -->
+```
+
+### Nuxt Data Fetching to Astro
+
+Nuxt has two methods of fetching server-side data:
+
+- [`asyncData` options API](https://nuxtjs.org/docs/features/data-fetching/#async-data)
+- [`fetch` hook](https://nuxtjs.org/docs/features/data-fetching/#the-fetch-hook)
+
+Astro simplifies this to simply fetch data inside of your page's code fence.
+
+Migrate the following:
+
+```vue title="pages/index.vue"
+{
+  // ...
+  async asyncData() {
+    const res = await fetch("https://pokeapi.co/api/v2/pokemon?limit=151")
+    const resJson = await res.json();
+    const pokemons = resJson.results;
+    return {
+      pokemons,
+    }
+  },
+}
+```
+
+To a code fence without a wrapper function:
+
+```astro title="src/pages/index.astro"
+---
+const res = await fetch("https://pokeapi.co/api/v2/pokemon?limit=151")
+const resJson = await res.json();
+const pokemons = resJson.results;
+---
+
+<!-- Your template here -->
+```
+
+### Nuxt Styling to Astro
```

Comment:
I would probably tighten this section up as these work the same? If we're focusing on what the migrating peep needs to do... then maybe it's nothing? In which case, I don't think we need a before/after comparison. Something more like just "both use style tags, so your existing Nuxt style tag doesn't have to change"

Reply:
I think it's worth showing the examples.
In Nuxt, the style tag lives outside of the template. In Astro, it's part of it. And you don't have the `scoped` keyword in Astro - it's automatically scoped. Seeing the examples can cement both of these points.